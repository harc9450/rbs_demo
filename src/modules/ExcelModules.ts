export interface IExcelColumns {
  name: string;
  key: number;
}

export interface IExcelDataToShow {
  rows: unknown[];
  cols: IExcelColumns[];
}
