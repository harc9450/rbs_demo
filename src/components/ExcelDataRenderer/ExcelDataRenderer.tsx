import React from "react";
import { IExcelDataToShow } from "../../modules/ExcelModules";

export const ExcelGrid = (props: IExcelDataToShow) => {
  const { rows, cols } = props;

  return (
    <div className="table table-responsive">
      <table className="table table-striped table-bordered">
        <tbody>
          {rows.map((value: any, index) => (
            <tr key={index}>
              {cols.map(column => {
                return column.name !== "A" && column.name !== "B" ? (
                  <td key={column.key}>{value[column.key]}</td>
                ) : null;
              })}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};
