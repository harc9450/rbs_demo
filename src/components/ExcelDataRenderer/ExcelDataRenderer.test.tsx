import * as React from "react";
import Enzyme from "enzyme";
import toJson from "enzyme-to-json";
import { IExcelDataToShow } from "../../modules/ExcelModules";
import { ExcelGrid } from "./ExcelDataRenderer";

describe("ExcelFile Component", () => {
  let wrapper: Enzyme.ShallowWrapper;
  let mountedWrapper: Enzyme.ReactWrapper<any, any>;
  let props: IExcelDataToShow;

  beforeEach(() => {
    props = {
      rows: [],
      cols: []
    };
  });

  it("should render correctly", () => {
    wrapper = Enzyme.shallow(<ExcelGrid {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should have a div element", () => {
    mountedWrapper = Enzyme.mount(<ExcelGrid {...props} />);
    expect(mountedWrapper.find("div")).toHaveLength(1);
  });

  it("should have a table element", () => {
    mountedWrapper = Enzyme.mount(<ExcelGrid {...props} />);
    expect(mountedWrapper.find("table")).toHaveLength(1);
  });

  it("should render table body", () => {
    mountedWrapper = Enzyme.mount(<ExcelGrid {...props} />);
    expect(mountedWrapper.find("tbody")).toHaveLength(1);
  });

  it("should not render table row while props having default value only", () => {
    mountedWrapper = Enzyme.mount(<ExcelGrid {...props} />);
    expect(mountedWrapper.find("tr")).toHaveLength(0);
  });

  it("should render table row while props having a record", () => {
    props = {
      rows: ["row1"],
      cols: [{ key: 1, name: "row 1" }]
    };
    mountedWrapper = Enzyme.mount(<ExcelGrid {...props} />);
    expect(mountedWrapper.find("tr")).toHaveLength(1);
  });
});
