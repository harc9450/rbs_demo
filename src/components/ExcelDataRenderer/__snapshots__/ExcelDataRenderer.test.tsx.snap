// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExcelFile Component should render correctly 1`] = `
<div
  className="table table-responsive"
>
  <table
    className="table table-striped table-bordered"
  >
    <tbody />
  </table>
</div>
`;
